hub/policy/src/integration-test/java/uk/gov/ida/integrationtest/hub/policy/apprule/AuthnRequestFromTransactionResourceIntegrationTest.java

```diff
@@ -182,21 +180,11 @@ public void getSignInProcessDto_shouldReturnSignInDetailsDto(){
         TestSessionResourceHelper.createSessionInIdpSelectedState(session, samlResponse.getIssuer(), idpEntityId, client,
                 buildUriForTestSession(IDP_SELECTED_STATE, session));
 
-        List<IdpConfigDto> expectedIdpConfigDtos = Arrays.asList(
-                IdpConfigDtoBuilder.anIdpConfigDto().build(),
-                IdpConfigDtoBuilder.anIdpConfigDto().build(),
-                IdpConfigDtoBuilder.anIdpConfigDto().build()
-        );
-
-        List<String> expectedEntityIds = IdpSelectedStateBuilder.anIdpSelectedState().build().getAvailableIdentityProviderEntityIds();
-
         Response response = getAuthRequestSignInProcess(session);
 
         assertThat(response.getStatus()).isEqualTo(Response.Status.OK.getStatusCode());
         AuthnRequestSignInDetailsDto entity = response.readEntity(AuthnRequestSignInDetailsDto.class);
         assertThat(entity.getRequestIssuerId()).isEqualTo(samlResponse.getIssuer());
-        assertThat(entity.getAvailableIdentityProviderEntityIds()).isEqualTo(expectedEntityIds);
```

Comment:
how come we can get rid of these assertions

Reply:
Becasue the AuthnRequestSignInDetailsDto object no longer has the methods/fields getAvailableIdentityProviderEntityIds() and getAvailableIdentityProviders(). They were not really used, and this information is now retrieved from the new config endpoint instead (using the proxy).

Reply:
perfect. Thanks

Reply:
It would be good to talk more about the difference between storing the IDPs in the session state vs dynamically getting them from the config service (and the relative merits)

Reply:
In general, we store these properties in the states themselves whenever we can. For instance, we can only get a list of available IDPs once we get to the IdpSelected stated, but not before, we don't have the required information to obtain that list. In this specific case, AuthnRequestSignInDetailsDto had this information stored but it was only used in an integration test and not anywhere in the code, so I removed it.

Reply:
As I understood it the available IDPs were cached in the session state when the session was created and Policy used that store directly afterwards (instead of Config), as well as storing the IDP that was selected.  Things must have changed since I last looked. 

FWIW saml-engine, however, uses IDP info directly from Config.